views/default/resources/favicon.ico.php

```diff
@@ -1,12 +1,20 @@
 <?php
-
 /**
  * Handle requests for /favicon.ico
  */
 
-elgg_set_http_header('Expires: ' . gmdate('D, d M Y H:i:s \G\M\T', strtotime("+1 week")), true);
-elgg_set_http_header("Pragma: public", true);
-elgg_set_http_header("Cache-Control: public", true);
+elgg_set_http_header('Expires: ' . gmdate('D, d M Y H:i:s \G\M\T', strtotime('+1 week')), true);
+elgg_set_http_header('Pragma: public', true);
+elgg_set_http_header('Cache-Control: public', true);
+
+$site = elgg_get_site_entity();
+if ($site->hasIcon('icon-32', 'favicon')) {
+	elgg_set_http_header('Content-Type: image/jpeg');
```

Comment:
yet another assumption ;)

Reply:
same as previous... generate icons can't be anything else then jpegs at the moment